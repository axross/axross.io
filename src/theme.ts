import Theme from './core/Theme';

const theme: Theme = {
  spacing: {
    small: 8,
    regular: 16,
    large: 24,
    huge: 40,
    enormous: 80,
  },
  typography: {
    body1: {
      size: 16,
      weight: 400,
    },
    body2: {
      size: 14,
      weight: 400,
    },
    heading1: {
      size: 20,
      weight: 400,
    },
    heading2: {
      size: 24,
      weight: 400,
    },
    heading3: {
      size: 45,
      weight: 900,
    },
    heading4: {
      size: 56,
      weight: 900,
    },
    heading5: {
      size: 112,
      weight: 900,
    },
  },
};

export default theme;
