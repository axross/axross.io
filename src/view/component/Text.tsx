import { ClassAttributes, createElement, ReactNode } from 'react';
import styled from '../../core/emotion';

type Type = 'body1' | 'heading1' | 'heading2' | 'heading3' | 'heading4' | 'heading5' | 'body2';

type Props = ClassAttributes<HTMLElement> & {
  type?: Type;
  sentence?: boolean;
  className?: string;
  children?: ReactNode;
};

const Text = ({ type = 'body1', sentence = false, className, children }: Props) => (
  <Root _type={type} sentence={sentence} className={className}>
    {children}
  </Root>
);

const TYPE: Record<Type, string> = {
  body1: `
    font-size: 16px;
    font-weight: 400;
  `,
  body2: `
    font-size: 14px;
    font-weight: 400;
  `,
  heading1: `
    font-size: 20px;
    font-weight: 400;
  `,
  heading2: `
    font-size: 24px;
    font-weight: 400;
  `,
  heading3: `
    font-size: 45px;
    font-weight: 900;
  `,
  heading4: `
    font-size: 56px;
    font-weight: 900;
  `,
  heading5: `
    font-size: 112px;
    font-weight: 900;
  `,
};

const Root = styled<{ _type: Type; sentence: boolean }, 'span'>('span')`
  display: block;
  line-height: ${({ sentence }) => (sentence ? 1.5 : 1)};
  color: #fff;
  font-family: Montserrat;
  -webkit-font-smoothing: antialiased;
  ${({ _type }) => TYPE[_type]};
`;

export default Text;
