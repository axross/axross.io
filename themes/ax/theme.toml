# theme.toml template for a Hugo theme
# See https://github.com/spf13/hugoThemes#themetoml for an example

name = "Ax"
license = "MIT"
licenselink = "https://github.com/axross/axross.io/blob/master/LICENSE.md"
description = ""
homepage = "http://axross.io/"
tags = []
features = []
min_version = 0.19

[author]
  name = "axross"
  homepage = "http://axross.io"
