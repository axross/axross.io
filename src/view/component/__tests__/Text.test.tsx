import { ThemeProvider } from 'emotion-theming';
import { createElement } from 'react';
import { create } from 'react-test-renderer';
import theme from '../../../theme';
import Text from '../Text';

describe('<Text>', () => {
  test('it matches with the previous snapshot (props.type="body1")', () => {
    expect(
      create(
        <ThemeProvider theme={theme}>
          <Text type="body1">text</Text>
        </ThemeProvider>,
      ).toJSON(),
    ).toMatchSnapshot();
  });

  test('it matches with the previous snapshot (props.type="body2")', () => {
    expect(
      create(
        <ThemeProvider theme={theme}>
          <Text type="body2">text</Text>
        </ThemeProvider>,
      ).toJSON(),
    ).toMatchSnapshot();
  });

  test('it matches with the previous snapshot (props.type="heading1")', () => {
    expect(
      create(
        <ThemeProvider theme={theme}>
          <Text type="heading1">text</Text>
        </ThemeProvider>,
      ).toJSON(),
    ).toMatchSnapshot();
  });

  test('it matches with the previous snapshot (props.type="heading2")', () => {
    expect(
      create(
        <ThemeProvider theme={theme}>
          <Text type="heading2">text</Text>
        </ThemeProvider>,
      ).toJSON(),
    ).toMatchSnapshot();
  });

  test('it matches with the previous snapshot (props.type="heading3")', () => {
    expect(
      create(
        <ThemeProvider theme={theme}>
          <Text type="heading3">text</Text>
        </ThemeProvider>,
      ).toJSON(),
    ).toMatchSnapshot();
  });

  test('it matches with the previous snapshot (props.type="heading4")', () => {
    expect(
      create(
        <ThemeProvider theme={theme}>
          <Text type="heading4">text</Text>
        </ThemeProvider>,
      ).toJSON(),
    ).toMatchSnapshot();
  });

  test('it matches with the previous snapshot (props.type="heading5")', () => {
    expect(
      create(
        <ThemeProvider theme={theme}>
          <Text type="heading5">text</Text>
        </ThemeProvider>,
      ).toJSON(),
    ).toMatchSnapshot();
  });

  test('it matches with the previous snapshot (props.type=undefined)', () => {
    expect(
      create(
        <ThemeProvider theme={theme}>
          <Text type="body1">text</Text>
        </ThemeProvider>,
      ).toJSON(),
    ).toMatchSnapshot();
  });

  test('it matches with the previous snapshot (props.type="heading2", props.sentence=true)', () => {
    expect(
      create(
        <ThemeProvider theme={theme}>
          <Text type="heading2" sentence>
            text
          </Text>
        </ThemeProvider>,
      ).toJSON(),
    ).toMatchSnapshot();
  });
});
