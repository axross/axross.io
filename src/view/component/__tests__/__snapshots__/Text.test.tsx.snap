// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text> it matches with the previous snapshot (props.type="body1") 1`] = `
.emotion-0 {
  display: block;
  line-height: 1;
  color: #fff;
  font-family: Montserrat;
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  font-weight: 400;
}

<span
  className="emotion-0"
>
  text
</span>
`;

exports[`<Text> it matches with the previous snapshot (props.type="body2") 1`] = `
.emotion-0 {
  display: block;
  line-height: 1;
  color: #fff;
  font-family: Montserrat;
  -webkit-font-smoothing: antialiased;
  font-size: 14px;
  font-weight: 400;
}

<span
  className="emotion-0"
>
  text
</span>
`;

exports[`<Text> it matches with the previous snapshot (props.type="heading1") 1`] = `
.emotion-0 {
  display: block;
  line-height: 1;
  color: #fff;
  font-family: Montserrat;
  -webkit-font-smoothing: antialiased;
  font-size: 20px;
  font-weight: 400;
}

<span
  className="emotion-0"
>
  text
</span>
`;

exports[`<Text> it matches with the previous snapshot (props.type="heading2") 1`] = `
.emotion-0 {
  display: block;
  line-height: 1;
  color: #fff;
  font-family: Montserrat;
  -webkit-font-smoothing: antialiased;
  font-size: 24px;
  font-weight: 400;
}

<span
  className="emotion-0"
>
  text
</span>
`;

exports[`<Text> it matches with the previous snapshot (props.type="heading2", props.sentence=true) 1`] = `
.emotion-0 {
  display: block;
  line-height: 1.5;
  color: #fff;
  font-family: Montserrat;
  -webkit-font-smoothing: antialiased;
  font-size: 24px;
  font-weight: 400;
}

<span
  className="emotion-0"
>
  text
</span>
`;

exports[`<Text> it matches with the previous snapshot (props.type="heading3") 1`] = `
.emotion-0 {
  display: block;
  line-height: 1;
  color: #fff;
  font-family: Montserrat;
  -webkit-font-smoothing: antialiased;
  font-size: 45px;
  font-weight: 900;
}

<span
  className="emotion-0"
>
  text
</span>
`;

exports[`<Text> it matches with the previous snapshot (props.type="heading4") 1`] = `
.emotion-0 {
  display: block;
  line-height: 1;
  color: #fff;
  font-family: Montserrat;
  -webkit-font-smoothing: antialiased;
  font-size: 56px;
  font-weight: 900;
}

<span
  className="emotion-0"
>
  text
</span>
`;

exports[`<Text> it matches with the previous snapshot (props.type="heading5") 1`] = `
.emotion-0 {
  display: block;
  line-height: 1;
  color: #fff;
  font-family: Montserrat;
  -webkit-font-smoothing: antialiased;
  font-size: 112px;
  font-weight: 900;
}

<span
  className="emotion-0"
>
  text
</span>
`;

exports[`<Text> it matches with the previous snapshot (props.type=undefined) 1`] = `
.emotion-0 {
  display: block;
  line-height: 1;
  color: #fff;
  font-family: Montserrat;
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  font-weight: 400;
}

<span
  className="emotion-0"
>
  text
</span>
`;
